docs/core-design.md

```diff
@@ -0,0 +1,354 @@
+        // +kubebuilder:default={{type: "Ready", status: "Unknown", reason:"Pending", message:"Waiting for controller", lastTransitionTime: "1970-01-01T00:00:00Z"}}
+        Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+##### Group
+
+This object will be in the `multinetwork.networking.k8s.io` API group. This will be a new group called `multinetwork`.
+
+#### Examples
+
+##### Generic
+
+```yaml
+apiVersion: multinetwork.networking.k8s.io/v1
+kind: PodNetwork
+metadata:
+  name: net-dataplane
+spec:
+  enabled: true
+  provider: "foo.io/bar"
+status:
+  conditions:
+  - lastProbeTime: null
+    lastTransitionTime: "2025-01-31T18:38:01Z"
+    status: "True"
+    type: Ready
+```
+
+##### With implementation-specific config
+
+```yaml
+apiVersion: multinetwork.networking.k8s.io/v1
+kind: PodNetwork
+metadata:
+  name: dataplane
+spec:
+  enabled: true
+  provider: "example.dra.x-k8s.io/network"
+  parameters:
+    master: "eth0"
+    mode: "bridge"
+    ipam:
+      type: "host-local"
+      subnet: "192.168.1.0/24"
+      rangeStart: "192.168.1.200"
+      rangeEnd: "192.168.1.216"
+      routes:
+      - dst: "0.0.0.0/0"
+      gateway: "192.168.1.1"
+status:
+  conditions:
+  - lastProbeTime: null
+    lastTransitionTime: "2022-11-17T18:38:01Z"
+    status: "True"
+    type: Ready
+```
+
+#### Status Conditions
+
+The PodNetwork object will use following conditions:
+
+* **Ready** \- indicates that the PodNetwork object is correct (validated) and can be consumed by the DRA Driver. New Pods should not be attached to a PodNetwork that is not Ready. This condition does not indicate readiness of specific PodNetwork on a per Node-basis. Following are the error reasons for this condition:
+
+
+| Reason name | Description |
+| ----- | ----- |
+| ConditionsNotReady | Other conditions on the object have “false” value. |
+| AdministrativelyDisabled | The PodNetwork’s Enabled field is set to false. |
+
+The provider’s implementation is responsible for the conditions life-cycle.
+
+##### Other Conditions
+
+We will not restrict adding other conditions to this object. This way implementations can further control the Readiness of PodNetwork by adding their own Condition and logic for it.
+
+##### Implementation consideration
+
+Implementations must enforce the Ready Condition from PodNetwork. It is up to the implementation for the solution, e.g. withdraw resource availability for the specific PodNetwork across Nodes.
+
+#### Provider
+
+The provider field gives the ability to uniquely identify what implementation (provider) is going to handle specific instances of PodNetwork objects. The value has to be in the form of an url. It is the implementer’s decision on how this field is going to be leveraged. They will decide how they behave when this field is empty and what specific value they are going to respect. This will dictate if a specific implementation can co-exist with other ones in the same cluster.
```

Comment:
other `implementer’s decision` field means no portability

Reply:
I really do not understand the goal of the API, it just has a boolean to enable or disable a "network" and passes an opaque config across all nodes ... in gateway there is a minimum of vendors that agree on and implement the API before is approved, do we have a consensus here with multiple implemnetations?